browser/renderer_context_menu/brave_spelling_options_submenu_observer.cc

```diff
@@ -0,0 +1,68 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/renderer_context_menu/brave_spelling_options_submenu_observer.h"
+
+#include "chrome/app/chrome_command_ids.h"
+#include "chrome/grit/generated_resources.h"
+#include "components/renderer_context_menu/render_view_context_menu_proxy.h"
+#include "ui/base/l10n/l10n_util.h"
+
+BraveSpellingOptionsSubMenuObserver::BraveSpellingOptionsSubMenuObserver(
+    RenderViewContextMenuProxy* proxy,
+    ui::SimpleMenuModel::Delegate* delegate,
+    int group_id)
+    : SpellingOptionsSubMenuObserver(proxy, delegate, group_id),
+      gtest_mode_(GTEST_MODE_DISABLED) {}
+
+void BraveSpellingOptionsSubMenuObserver::InitMenu(
+    const content::ContextMenuParams& params) {
+  // Let Chromium build the submenu.
+  SpellingOptionsSubMenuObserver::InitMenu(params);
+
+  // Assumptions:
+  // 1. Use of spelling service is disabled in Brave profile preferences.
+  // 2. We overrode RenderViewContextMenu::AddSpellCheckServiceItem so that the
+  // spelling suggestions toggle isn't added to the menu by the base class.
+  DCHECK(!use_spelling_service_.GetValue());
+  DCHECK(submenu_model_.GetIndexOfCommandId(
+             IDC_CONTENT_CONTEXT_SPELLING_TOGGLE) == -1);
+
+  // Check if we ended up with a separator as the last item and, if so, get rid
+  // of it.
+  int index = submenu_model_.GetItemCount() - 1;
+  if (index >= 0 &&
+      submenu_model_.GetTypeAt(index) == ui::MenuModel::TYPE_SEPARATOR)
+    submenu_model_.RemoveItemAt(index);
+  DCHECK(submenu_model_.GetItemCount());
+
+  // Special accommodations for gtest.
+  if (gtest_mode_ != GTEST_MODE_DISABLED) {
```

Comment:
Just a thought for the future (I'm curious too) - you might check if unit/browser tests expose a `#define` that you can check for with preprocessors? ex: you could wrap this section (and potentially the definition/setting of `gtest_mode_`) with `#ifdef`/`#endif` guards

Reply:
I was looking for something like this as well but didn't see any additional definitions in the tests build in our build setup. gtest itself doesn't seem to add any (https://stackoverflow.com/questions/6077833/how-to-check-if-google-test-is-running-in-my-code). Ideally, I wouldn't want to have a different code path for tests in the first place, but in this case it would mean creating a more complex mock object (the current one was taken from Chromium and slightly modified).